course4/homework-string/string.cc

```diff
@@ -4,6 +4,129 @@ namespace course {
 
 string::string()
 {
+	mStr = std::make_unique<char[]>(1);
+	mStr[0] = NULL;
+	mStrSize = 0;
 }
 
-}
\ No newline at end of file
+string::string(size_t n)
+{
+	mStr = std::make_unique<char[]>(n + 1);
+	mStr[n] = NULL;
+	mStrSize = n;
+}
+
+string::string(const char* s1)
+{
+	mStrSize = strlen(s1);
```

Comment:
Use ```std::strlen``` instead of ```strlen```. This also applies to other library functions: prefer C++ functions over the C ones.